qa/ccs-common-rest/build.gradle

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+
+import org.elasticsearch.gradle.Version
+import org.elasticsearch.gradle.internal.test.RestIntegTestTask
+import org.elasticsearch.gradle.testclusters.DefaultTestClustersTask
+
+
+apply plugin: 'elasticsearch.internal-testclusters'
+apply plugin: 'elasticsearch.standalone-rest-test'
+apply plugin: 'elasticsearch.rest-resources'
+
+// This subproject copies a subset of the rest-api-spec rest tests and runs them in a slightly
+// modified way on two clusters connected via CCS. All operations except searches and a few other
+// APIs that support CCS are run agains the "write" cluster, so all indexed data will live there.
+// All search requests however are run against the other cluster connected via CCS to the "write"
+// cluster. The test runner modifies index names on these API calls to route to the remote cluster
+// and also modifies certain "match" sections to expect index names with the remote cluser prefix
+// on the fly while running these tests.
+
+restResources {
+  restApi {
+    include '_common', 'bulk', 'count', 'cluster', 'index', 'indices', 'field_caps', 'msearch',
+      'search', 'async_search', 'graph', '*_point_in_time', 'info', 'scroll', "clear_scroll"
+  }
+  restTests {
+    includeCore 'field_caps', 'msearch', 'search', 'suggest', 'scroll', "indices.resolve_index"
+  }
+}
+
+def writeCluster = testClusters.register("ccs-write") {
+  numberOfNodes = 2
+  setting 'node.roles', '[data,ingest,master]'
+}
+
+def searchCluster = testClusters.register("ccs-search") {
+  setting 'node.roles', '[data,ingest,master,remote_cluster_client]'
+  setting 'cluster.remote.connections_per_cluster', '1'
+  setting 'cluster.remote.remote_cluster.seeds',
+    { "\"${writeCluster.get().getAllTransportPortURI().get(0)}\"" }
+}
+
+testClusters.configureEach {
+  setting 'xpack.security.enabled', 'false'
+  requiresFeature 'es.index_mode_feature_flag_registered', Version.fromString("8.0.0")
+}
+
+// the following task is needed to make sure the write cluster is running before the search cluster
+// gets configured with the remotes cluster seed
+tasks.register('startWriteCluster', DefaultTestClustersTask) {
```

Comment:
We can rename them remote and local, I mostly changed the name because I got confused sometimes which one we "write" e.g. index to and which one we search agains.
The mapping would be:

local -> search 
remote -> write

I can change the names at least in the gradle file if you think it makes things clearer, then I will probably extend the explanation part a bit

Reply:
I already did the renaming, but I left the System property we get the client connection details from in CcsCommonYamlTestSuiteIT as "tests.rest.search_cluster" since in that test runner I'd like to make sure through variable naming that the client we create there in addition to the "main" test client is the one running the search APIs etc... Hope that makes sense